hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenUnfreezeAccountHandler.java

```diff
@@ -127,6 +131,8 @@ private TokenRelation validateSemantics(
         final var tokenRel = tokenRelStore.getForModify(accountId, tokenId);
         validateTrue(tokenRel != null, TOKEN_NOT_ASSOCIATED_TO_ACCOUNT);
 
+        // Check that token is not deleted
+        validateFalse(token != null && token.deleted(), TOKEN_WAS_DELETED);
```

Comment:
This may create another issue by combining two checks.
A possible improvement:
```suggestion
        validateFalse(token == null, /*Fill in appropriate code here*/);
        validateFalse(token.deleted(), TOKEN_WAS_DELETED);
```

Reply:
Agree, added `if` because I'm not sure if there is appropriate response code for non existing token.